<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>584</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="mainTabs">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>751</width>
      <height>501</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Настройка</string>
     </attribute>
     <widget class="QComboBox" name="portsComboBox">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>40</y>
        <width>121</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="baudRateLabel">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>20</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Введите Baud Rate </string>
      </property>
     </widget>
     <widget class="QLabel" name="portsLabel">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>20</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Выберите COM-порт</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="baudRateLineEdit">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>40</y>
        <width>113</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>57600</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="consoleEdit">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>711</width>
        <height>311</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
      </property>
     </widget>
     <widget class="QLabel" name="consoleLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>120</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Консоль:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_refresh">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>91</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Обновить</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="btn_radio_no_confirm">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>291</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Отключить подтверждение действий</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_dump_codes">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>80</y>
        <width>301</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Выгрузить G-коды для текущего рисунка</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_chk_connect">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>40</y>
        <width>181</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Проверить соединение</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Рисование командами</string>
     </attribute>
     <widget class="QGraphicsView" name="paint_view">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>40</y>
        <width>330</width>
        <height>228</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>330</width>
        <height>228</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>330</width>
        <height>228</height>
       </size>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>10</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Картинка (можно тыкать):</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinbox_paint_degree">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>250</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="maximum">
       <number>315</number>
      </property>
      <property name="singleStep">
       <number>45</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>331</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Используйте рисование/перемещение через виджет:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_paint_vertical">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>[ | ] Вертикальное перемещение</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_paint_horizontal">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>[ - ] Горизонтально перемещение</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_paint_sloped">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>[ \ ] Произвольное перемещение</string>
      </property>
     </widget>
     <widget class="QLabel" name="mode_paint_label">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>280</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>current_mode</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>250</y>
        <width>231</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Угол в градусах:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinbox_paint_len">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>280</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>500</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>280</y>
        <width>231</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Длина прямой (по x или y):</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>310</y>
        <width>231</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Радиус окружности:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinbox_paint_radius">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>310</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>500</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_paint_execute_command">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>310</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Выполнить команду</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="btn_radio_paint_ccw">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>370</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Дуга против часовой стрелки</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>340</y>
        <width>231</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Начальный угол дуги в градусах:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinbox_paint_degree_arc">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>340</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="maximum">
       <number>315</number>
      </property>
      <property name="singleStep">
       <number>45</number>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_paint_hatch">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>[ /// ] Заштриховать область</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="btn_radio_paint_ccw_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Рисование</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>220</y>
        <width>331</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Или введите настройки вручную:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_paint_hatch_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>160</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>[ ( ] Перемещение по дуге</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>440</y>
        <width>711</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Примечание: для рисования дуги все равно нужно задать настройки углов и радиуса вручную.</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab2">
     <attribute name="title">
      <string>Рисование в G-кодах</string>
     </attribute>
     <widget class="QGraphicsView" name="field_view">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>40</y>
        <width>330</width>
        <height>228</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>330</width>
        <height>228</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>330</width>
        <height>228</height>
       </size>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_g00">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>G00 Быстрое позиционирование</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_g01">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>G01 Линейная интерполяция</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_g02">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>G02 Круговая интерполяция по часовой стрелке</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_g03">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>G03 Круговая интерполяция против часовой стрелки</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_g90">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>180</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>G90 Абсолютная система координат</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_g91">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>210</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>G91 Относительная система координат</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_colebrate">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>290</y>
        <width>151</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Калибровка (на 0,0)</string>
      </property>
     </widget>
     <widget class="QLabel" name="mode_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>260</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>current_mode</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_clear_img">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>330</y>
        <width>151</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Очистить картинку</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="radiusLineEdit">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>330</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="radius_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>300</y>
        <width>131</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Радиус (для G02 и G03):</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>331</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Выбор G-кода:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>10</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Нажимайте точку в этом виджете:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>280</y>
        <width>331</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Или введите координаты вручную:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_x">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>310</y>
        <width>121</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_execute_command">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>350</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Выполнить команду</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>406</x>
        <y>312</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_y">
      <property name="geometry">
       <rect>
        <x>610</x>
        <y>310</y>
        <width>121</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>596</x>
        <y>312</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>380</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Прочитать из файла:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_select_file">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>410</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Выбрать файл</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_filename">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>410</y>
        <width>581</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Путь к файлу: файл не выбран.</string>
      </property>
     </widget>
     <widget class="QPushButton" name="push_file_send">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>440</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Отправить</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Режим отладки</string>
     </attribute>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>331</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Введите G-код:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_pro_gcode">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>561</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_pro_gcode_send">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>40</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Отправить</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>331</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Введите HEX отправляемого пакета на ЧПУ:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_pro_hex">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>561</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_pro_hex_send">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>110</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Отправить</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
